saleor/graphql/checkout/mutations/checkout_note_update.py

```diff
@@ -0,0 +1,53 @@
+import graphene
+
+from ....webhook.event_types import WebhookEventAsyncType
+from ...core import ResolveInfo
+from ...core.descriptions import ADDED_IN_321
+from ...core.doc_category import DOC_CATEGORY_CHECKOUT
+from ...core.mutations import BaseMutation
+from ...core.types import CheckoutError
+from ...core.utils import WebhookEventInfo
+from ...plugins.dataloaders import get_plugin_manager_promise
+from ..types import Checkout
+
+
+class CheckoutNoteUpdate(BaseMutation):
+    checkout = graphene.Field(Checkout, description="An updated checkout.")
+
+    class Arguments:
+        id = graphene.ID(
+            description="The checkout's ID." + ADDED_IN_321,
+            required=False,
+        )
+        note = graphene.String(required=True, description="New note content.")
```

Comment:
One more request, could you rename this input field with `customer_note`, we want to unify it with order mutations that are using `customerNote` 🙏

Reply:
Do you think I should also rename the mutation to `checkoutCustomerNoteUpdate`? I Was thinking about this and the naming of the `note` fields is currently quite confusing. 

In checkout, there is `note` field, that gets renamed to `customer_note` when checkout is converted to the order.

At the same time, there are new `orderNoteAdd` and `orderNoteUpdate` mutation that updates the notes in the order's `events` list. And there is actually no mutation to update the `customer_note` on the order object.

I was thinking about opening a PR to also implementing mutation on the order object but I was not sure about the correct way to do this, because there is an `orderUpdate` mutation that I could add this field to, but at the same time I think you are recently trying to do more specific update mutations that update single thing on the object instead of one big update mutation.

Reply:
@pitkes22 great points! Thank you <3

Yes, I think we can rename it to `checkoutCustomerNoteUpdate`, as it could be misleading with `orderNoteAdd` and `orderNoteUpdate` that are creating the staff notes.
And as you correctly noticed, we are also missing a mutation for customer note update on orders.

We are planning to deprecate `note` on `Checkout` type and add a `customerNote` field instead, to make it clear.